contracts/PoolManager.sol

```diff
@@ -338,26 +337,26 @@ contract PoolManager is IPoolManager, Fees, NoDelegateCall, ERC1155, IERC1155Rec
         _accountDelta(currency, -(paid.toInt128()));
     }
 
-    function _burnAndAccount(Currency currency, uint256 amount) internal {
-        _burn(address(this), currency.toId(), amount);
-        _accountDelta(currency, -(amount.toInt128()));
-    }
-
-    function onERC1155Received(address, address, uint256 id, uint256 value, bytes calldata) external returns (bytes4) {
-        if (msg.sender != address(this)) revert NotPoolManagerToken();
-        _burnAndAccount(CurrencyLibrary.fromId(id), value);
-        return IERC1155Receiver.onERC1155Received.selector;
+    /// @inheritdoc IPoolManager
+    function burn(Currency currency, uint256 amount) external noDelegateCall onlyByLocker {
+        _burnAndAccount(currency, amount);
     }
 
-    function onERC1155BatchReceived(address, address, uint256[] calldata ids, uint256[] calldata values, bytes calldata)
+    /// @inheritdoc IPoolManager
+    function batchBurn(Currency[] calldata currencies, uint256[] calldata amounts)
         external
-        returns (bytes4)
+        noDelegateCall
+        onlyByLocker
     {
-        if (msg.sender != address(this)) revert NotPoolManagerToken();
-        for (uint256 i; i < ids.length; i++) {
-            _burnAndAccount(CurrencyLibrary.fromId(ids[i]), values[i]);
+        uint256 length = currencies.length;
+        for (uint256 i; i < length; i++) {
+            _burnAndAccount(currencies[i], amounts[i]);
         }
-        return IERC1155Receiver.onERC1155BatchReceived.selector;
+    }
+
+    function _burnAndAccount(Currency currency, uint256 amount) internal {
+        _accountDelta(currency, -(amount.toInt128()));
+        _burn(currency.toId(), amount);
```

Comment:
insane nit, but the fn name <> op ordering should match? so do the burn first hahah